shell/platform/windows/keyboard_key_embedder_handler.cc

```diff
@@ -264,8 +264,17 @@ void KeyboardKeyEmbedderHandler::KeyboardHookImpl(
     pressingRecords_[physical_key] = eventual_logical_record;
   } else {
     auto record_iter = pressingRecords_.find(physical_key);
-    assert(record_iter != pressingRecords_.end());
-    pressingRecords_.erase(record_iter);
+    // Critical keys can be unset before this poitn if get_key_state_ finds they
```

Comment:
```suggestion
    // Critical keys can be unset before this point if get_key_state_ finds they
```

Reply:
I don't think we need to add this comment. The iterator should always be valid if all our other logic is correct. Since we've fixed the bug, this comment will no longer be relavent.

Reply:
I did more digging and the issue is not just that the scancode is 0, but rather that the `get_key_state_` method never reports that the caps lock key is pressed when Narrator issues its caps lock key event, which causes the caps lock entry in `pressingRecords_` to be unset earlier by `SynchronizeCritialPressedStates`.